### PR TITLE
OAuth: Remove the code to fetch an OAuth signup URL

### PR DESCRIPTION
**Pull request by scruffian**
Once #17530 is merged we can remove calls to get the OAuth signup URL.

#### Testing instructions
  
1. Run `git checkout update/remove-oauth-signup-link` and start your server
2. Open the [`WPCC` login page](http://calypso.localhost:3000/log-in?client_id=390)
3. Check that there are no requests to /oauth2/signup-url/
  
#### Reviews
  
- [x] Code
- [x] Product

### CODE REVIEW

File: client/blocks/login/login-form.jsx [hunk 1]
```diff
@@ -166,15 +160,11 @@ export class LoginForm extends Component {
...
-		if ( isOauthLogin ) {
-			if ( config.isEnabled( 'signup/wpcc' ) ) {
-				signupUrl = '/start/wpcc?' + qs.stringify( {
-					oauth2_client_id: oauth2Client.id,
-					oauth2_redirect: redirectTo
-				} );
-			} else {
-				signupUrl = oauth2Client.signupUrl;
-			}
+		if ( isOauthLogin && config.isEnabled( 'signup/wpcc' ) ) {
```
Comment (Tug) [inline client/blocks/login/login-form.jsx]: We should do something about this feature, either remove this check here or set `signup/wpcc: true` in all envs

**Comment by Tug** [comment]
Code 👍

File: client/blocks/login/login-form.jsx (hunk 1 above)
Comment (Tug) [reply client/blocks/login/login-form.jsx]: Oh it's handled in #17530 nvm then

**Comment by hafizrahman** [comment]
Product 👍

**Pull request merged**